homeassistant/components/metoffice/__init__.py

```diff
@@ -41,23 +42,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     if metoffice_data.site_name is None:
         raise ConfigEntryNotReady()
 
-    metoffice_coordinator = DataUpdateCoordinator(
+    metoffice_daily_coordinator = DataUpdateCoordinator(
         hass,
         _LOGGER,
-        name=f"MetOffice Coordinator for {site_name}",
-        update_method=metoffice_data.async_update,
+        name=f"MetOffice Daily Coordinator for {site_name}",
+        update_method=metoffice_data.async_update_daily,
+        update_interval=DEFAULT_SCAN_INTERVAL,
+    )
+
+    metoffice_hourly_coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=f"MetOffice Hourly Coordinator for {site_name}",
+        update_method=metoffice_data.async_update_hourly,
         update_interval=DEFAULT_SCAN_INTERVAL,
     )
 
     metoffice_hass_data = hass.data.setdefault(DOMAIN, {})
     metoffice_hass_data[entry.entry_id] = {
         METOFFICE_DATA: metoffice_data,
-        METOFFICE_COORDINATOR: metoffice_coordinator,
+        METOFFICE_HOURLY_COORDINATOR: metoffice_hourly_coordinator,
+        METOFFICE_DAILY_COORDINATOR: metoffice_daily_coordinator,
         METOFFICE_NAME: site_name,
     }
 
     # Fetch initial data so we have data when entities subscribe
-    await metoffice_coordinator.async_refresh()
+    await metoffice_hourly_coordinator.async_refresh()
+    await metoffice_daily_coordinator.async_refresh()
```

Comment:
Do you think it's worth using `asyncio.gather()` to run them in parallel?

Reply:
Done